contracts/ics/conn/src/entry.rs

```diff
@@ -26,25 +29,34 @@ pub fn main() -> Result<()> {
         | MsgType::MsgChannelOpenTry
         | MsgType::MsgChannelOpenAck
         | MsgType::MsgChannelOpenConfirm => return Ok(()),
+        MsgType::MsgClientCreate => return check_create(),
         _ => return Err(Error::UnexpectedMsg),
     }
 
-    let _current_script = hl::load_script().map_err(|_| Error::LoadScriptErr)?;
-    let client_data = hl::load_cell_data(0, Source::CellDep).map_err(|_| Error::LoadCellDataErr)?;
-    let client = decode::<Client>(&client_data).map_err(|_| Error::ClientEncoding)?;
+    let client = load_client()?;
+
+    let new_lock = hl::load_cell_lock(0, Source::Output).map_err(|_| Error::ConnectionLock)?;
+    let new_lock_args = new_lock.args();
+    let new_connection_args =
+        ConnectionArgs::from_slice(new_lock_args.as_slice()).map_err(|_| Error::ConnectionLock)?;
```

Comment:
I suggest wrapping them into a utility function to enhance the code understanding